app/policies/user_policy.rb

```diff
@@ -0,0 +1,40 @@
+class UserPolicy < ApplicationPolicy
+  class Scope < Scope
+    def resolve
+      scope.joins(:memberships).where(memberships: { organisation: user.organisations })
+    end
+  end
+
+  def show?
+    is_user || user_belongs_to_same_organisation
+  end
+
+  def create?
```

Comment:
Shouldn't this also be scoped to an admin of the organisation? Or maybe `super_admin` in this case? But if you're an admin of a law_firm, then you shouldn't be able to create a use within cso.